Source/WebCore/rendering/RenderListMarker.cpp

```diff
@@ -1846,12 +1846,14 @@ void RenderListMarker::updateContent()
         m_textIsLeftToRightDirection = u_charDirection(m_textWithSuffix[0]) != U_RIGHT_TO_LEFT;
         break;
     case ListStyleType::Type::CounterStyle: {
-        auto* counter = counterStyle();
+        auto counter = counterStyle();
         ASSERT(counter);
-        auto text = makeString(counter->prefix(), counter->text(m_listItem->value()));
-        m_textWithSuffix = makeString(text, counter->suffix());
-        m_textWithoutSuffixLength = text.length();
-        m_textIsLeftToRightDirection = u_charDirection(text[0]) != U_RIGHT_TO_LEFT;
+        if (counter) {
```

Comment:
Why do we need this new check? `counterStyle()` should always return a counter style in this case.

Reply:
It should, but since the map is populated dynamically, if something that we aren't predicting goes wrong, I think it is more prudent to assert + avoid the nullptr deref than just asserting.

Reply:
If we make `resolvedCounterStyle` return a `Ref<>`, I think it should give us this guarantee? If something goes wrong, we should fix it. We don't want to hide a nullptr bug behind a visual bug (no marker shown instead of decimal/bullets).

Reply:
I agree we shouldn't hide it, that's why I'm asserting, but I'd say it is better to fail an assert in a debug build with an eventual problem than derefing a nullptr.

Reply:
The ASSERT is fine, but I think we should remove the  `if (counter)` check.